ir/x86_intrinsics.h

```diff
@@ -0,0 +1,141 @@
+#pragma once
+
+#include "ir/instr.h"
+
+namespace IR {
+class FakeShuffle final : public Instr {
+  Value *v1, *v2, *mask;
+
+public:
+  FakeShuffle(Type &type, std::string &&name, Value &v1, Value &v2, Value &mask)
+      : Instr(type, std::move(name)), v1(&v1), v2(&v2), mask(&mask) {}
+  std::vector<Value *> operands() const override;
+  bool propagatesPoison() const override;
+  bool hasSideEffects() const override;
+  void rauw(const Value &what, Value &with) override;
+  void print(std::ostream &os) const override;
+  StateValue toSMT(State &s) const override;
+  smt::expr getTypeConstraints(const Function &f) const override;
+  std::unique_ptr<Instr> dup(Function &f,
+                             const std::string &suffix) const override;
+};
+
+class X86IntrinBinOp final : public Instr {
+public:
+  static constexpr unsigned numOfX86Intrinsics = 135;
```

Comment:
this constant shouldn't exist; it is a maintenance headache. the arrays below have the size anyway.